Test data integrity

nitisht opened the pull request:
- Send logs to Parseable
- Wait for sync
- Download parquet from the store created by Parseable for the minute
- Compare the sent logs with the ones loaded from the downloaded parquet

nitisht commented:
I have read the CLA Document and I hereby sign the CLA

nitisht commented:
cc @nikhil-bijapur can you sign the CLA with the comment

theteachr commented:
I have read the CLA Document and I hereby sign the CLA

nitisht commented:
recheck

theteachr committed:
Test data integrity

- Send logs to Parseable
- Wait for sync
- Download parquet from the store created by Parseable for the minute
- Compare the sent logs with the ones loaded from the downloaded parquet

nitisht committed:
Add flags to main shell script so it is passed on to go code

theteachr committed:
:recycle: Use listing instead of prefix generation

- Generate logs dynamically using `flog`
- Ingest across minute boundaries

theteachr commented:
- [ ] Mark test failure after `slog.Error` calls

theteachr committed:
:truck: Use a better name

nitisht commented on integrity_test.go:
```diff
@@ -0,0 +1,256 @@
+			continue
+		}
+
+		parquetObject, err := client.GetObject("integrity-test", key, minio.GetObjectOptions{})
```
Dont hard code the bucket name here

theteachr committed:
Fix hardcoded bucket name

trueleo commented on integrity_test.go:
```diff
@@ -0,0 +1,256 @@
+		// store.
+		actualFlog := actualFlogs[rowCount-i-1].Deref()
+		require.Equal(t, actualFlog, expectedFlog)
+	}
```
I think here the stream can safely deleted after the test has finished. Otherwise LGTM :+1:

nitisht commented:
@trueleo have you tested? does it work as expected

trueleo commented:
@theteachr It seems to be failing with 

```
docker run --network=host trueleo8/quest:latest smoke http://localhost:8000 admin admin 
=== RUN   TestIntegrity
--- FAIL: TestIntegrity (120.11s)
2023/12/19 12:21:54 ERROR couldn't create MinIO client error="address http://localhost:9000: too many colons in address"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7b5822]

goroutine 19 [running]:
testing.tRunner.func1.2({0x8025a0, 0xbac4d0})
        /usr/local/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x8025a0?, 0xbac4d0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
quest.downloadParquetFiles({0x7ffdf254beab, 0xa}, {{0x7ffdf254bed9, 0x15}, {0x7ffdf254befb, 0xa}, {0x7ffdf254bf12, 0xa}, {0x7ffdf254bf2b, 0x9}})
        /tests/integrity_test.go:155 +0x102
quest.TestIntegrity(0x0?)
        /tests/integrity_test.go:121 +0x47b
testing.tRunner(0xc000107380, 0x8c8670)
        /usr/local/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 1
        /usr/local/go/src/testing/testing.go:1648 +0x3ad
```

I am running with default params and created a bucket named `parseable` . There is some data in that bucket now, so it seems ingest step worked but afterward that it failed on the next step.

theteachr commented on main.sh:
```diff
@@ -26,11 +26,19 @@ vus=$6
 duration=$7
 : "${duration:="5m"}"
+minio_url=$8
+: "${minio_url:="http://localhost:9000"}"
```
```suggestion
: "${minio_url:="localhost:9000"}"
```
@trueleo This should fix https://github.com/parseablehq/quest/pull/47#issuecomment-1862669882.

theteachr committed:
Delete stream at the end of test

theteachr committed:
Fix `minio_url`

theteachr committed:
Use the same default for bucket name

theteachr committed:
Don't run ingestion inside a Go routine

Doing so was causing the wait time to be less than the sync time.

* Misc Updates *
- Check ingest response from Parseable
- Add logs
